homeassistant/components/surepetcare/sensor.py

```diff
@@ -98,3 +102,37 @@ async def async_added_to_hass(self) -> None:
             async_dispatcher_connect(self.hass, TOPIC_UPDATE, self._async_update)
         )
         self._async_update()
+
+
+class SureLock(SensorEntity):
+    """A sensor implementation for Sure Petcare Entities."""
+
+    _attr_should_poll = False
+
+    def __init__(self, _id: int, spc: SurePetcareAPI) -> None:
+        """Initialize a Sure Petcare sensor."""
+        self._id = _id
+        self._spc: SurePetcareAPI = spc
+
+        surepy_entity: SurepyEntity = self._spc.states[_id]
+
+        self._attr_device_class = None
+        self._attr_name = f"{surepy_entity.type.name.capitalize()} {surepy_entity.name.capitalize()} Lock"
+        self._attr_unique_id = f"{surepy_entity.household_id}-{surepy_entity.id}-lock"
+
+    @callback
+    def _async_update(self) -> None:
+        """Get the latest data and update the state."""
+        surepy_entity = self._spc.states[self._id]
+        state = surepy_entity.state
+        self._attr_state = state
```

Comment:
What values can the lock state take?

Reply:
Their listed in the code elsewhere as an enumeration. That said I'm withdrawing this pull request, since the integration is missing too much functionality (e.g. it only updates when you restart HA). There is an alternate available via HACS which seems more full-featured.